### PR TITLE
Add inner wrapper div in 'td.wc-block-cart-item__product'

### PR DESCRIPTION
**Pull request by manospsyx**
Adding this wrapper element is necessary in order for us to be able to "fake" item grouping in a theme-agnostic manner: 

https://user-images.githubusercontent.com/1783726/118831630-97855600-b8c8-11eb-8906-c7f9ada769e7.png

Discussed in #4241 

This is a CSS requirement -- it is not possible to reliably obtain the result seen above (let alone adjust it via breakpoints) without a wrapper element.

#### Other Checks

- [ ] I've updated [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) for any feature flags or experimental interfaces implemented in this pull request.
- [ ] I tagged two reviewers because this PR makes queries to the database or I think it might have some security impact.

#### Automated Tests
* [x] Changes in this PR are covered by Automated Tests.
  * [x] Unit tests
  * [x] E2E tests

### Manual Testing

How to test the changes in this Pull Request:

1. Add two different products to the cart.
2. Inspect the cart table when the small/medium/large CSS breakpoints are active.
3. Confirm that no visual issues are caused by the additional `div`.

### User Facing Testing

These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).
* [x] Same as above, or
* [ ] See steps below.

1.
2.
3.

### Performance Impact

None

### Changelog

> Wrapped cart item product contents in inner div.

### CODE REVIEW

File: assets/js/blocks/cart-checkout/cart/cart-line-items-table/cart-line-item-row.tsx
Comment (opr) [inline assets/js/blocks/cart-checkout/cart/cart-line-items-table/cart-line-item-row.tsx]: I think we should remove `classnames` here _or_ add a filter to supply a custom wrapper class name. (Don't think there would be a good case for doing this so I'm leaning toward just dropping it)

**Review by opr — CHANGES_REQUESTED** [review]
Hey Manos, I think this is fine, I see Nadir raised some points in Slack p1642674032008700-slack-C02TS23QJ1X about testing it when the basket is displayed as a grid, this happens when the screen is medium or smaller and from my testing I can't see any issues with adding this wrapper, it seems like it would unblock you quite well.

There are a few eslint errors but this just refers `QuantitySelector` on 290-308, it needs to be indented by one tab.

Also before merging this PR, some testing instructions for when we write release notes would be good.

**Comment by manospsyx** [comment]
Hey @opr --

> from my testing I can't see any issues with adding this wrapper

Awesome, that's in line with what I saw when I tested this a while back.

I have removed `classnames` call as suggested, but can't quite get how to properly indent the code. I pretty much took the inner code from `trunk` and added the wrapper element around it after indenting by 1 tab.

> some testing instructions for when we write release notes would be good.

I didn't add any tests here as I didn't anticipate any breaking changes initially -- I added some instructions in line with Nadir's suggestions (test at each of the 3 screen sizes for which CSS breakpoints exist). Let me know if this is okay!

**Comment by opr** [comment]
> Hey @opr --
> 
> > from my testing I can't see any issues with adding this wrapper
> 
> Awesome, that's in line with what I saw when I tested this a while back.
> 
> I have removed `classnames` call as suggested, but can't quite get how to properly indent the code. I pretty much took the inner code from `trunk` and added the wrapper element around it after indenting by 1 tab.
> 
> > some testing instructions for when we write release notes would be good.
> 
> I didn't add any tests here as I didn't anticipate any breaking changes initially -- I added some instructions in line with Nadir's suggestions (test at each of the 3 screen sizes for which CSS breakpoints exist). Let me know if this is okay!

Ace! No need for automated tests here, I just meant manual steps in the PR body, thanks for adding them! Just so you know I asked for this because when we come to release Blocks we copy the testing instructions from each PR into the release notes, and a third party will test it during the WooCommerce core release cycle 😎

**Review by opr — DISMISSED** [review]

**Comment by opr** [comment]
😖  sorry I approved too early, would you like me to fix the indentation issue on this branch for you (once this is fixed it'll be ok to merge)

**Comment by manospsyx** [comment]
> would you like me to fix the indentation issue on this branch for you

Yes, please! I'm curious to understand exactly what the linter expects here. Thank you, Thomas!

**Review by opr — APPROVED** [review]
I updated the indentation, so besides this the changes look fine and work well.

Thanks Manos! 😁

**Pull request merged**
